cognite/neat/_graph/extractors/_classic_cdf/_base.py

```diff
@@ -182,9 +223,27 @@ def _get_rdf_type(self, item: T_CogniteResource) -> str:
 
     def _as_object(self, raw: Any, key: str) -> Literal | URIRef:
         if key in {"data_set_id", "dataSetId"}:
-            return self.namespace[f"{InstanceIdPrefix.data_set}{raw}"]
+            if self.identifier == "externalId" and self.lookup_dataset_external_id:
+                try:
+                    data_set_external_id = self.lookup_dataset_external_id(raw)
+                except KeyError:
+                    return Literal("Unknown data set", datatype=XSD.string)
```

Comment:
```suggestion
                    return Literal("Unknown data set")
```
it will automatically pick up xsd type